pytomorrowio/fields.py

```diff
@@ -0,0 +1,232 @@
+"""Tomorrow.io field definitions"""
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import List
+
+from .const import (
+    ONE_DAY,
+    ONE_HOUR,
+    REALTIME,
+    THIRTY_MINUTES,
+    TYPE_AIR_QUALITY,
+    TYPE_FIRE,
+    TYPE_POLLEN,
+    TYPE_PRECIPITATION,
+    TYPE_SOLAR,
+    TYPE_WEATHER,
+)
+
+MIN = "Min"
+MAX = "Max"
+AVG = "Avg"
+
+ALL_MEASUREMENTS = [MIN, MAX, AVG]
+NO_AVG = [MIN, MAX]
+
+
+@dataclass
+class FieldDefinition:
+    """Tomorrow.io field definition"""
+
+    max_timestep: timedelta
+    measurements: List[str]
+    type: str
+
+
+FIELDS_V4 = {
+    "temperature": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "temperatureApparent": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "dewPoint": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "humidity": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "windSpeed": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "windDirection": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=[AVG],
+        type=TYPE_WEATHER,
+    ),
+    "windGust": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "pressureSurfaceLevel": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "pressureSeaLevel": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_WEATHER,
+    ),
+    "precipitationIntensity": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_PRECIPITATION,
+    ),
+    "precipitationProbability": FieldDefinition(
+        max_timestep=ONE_DAY,
+        measurements=ALL_MEASUREMENTS,
+        type=TYPE_PRECIPITATION,
+    ),
+    "precipitationType": FieldDefinition(
+        max_timestep=ONE_HOUR,
+        measurements=[],
+        type=TYPE_PRECIPITATION,
+    ),
+    "hailBinary": FieldDefinition(
+        max_timestep=REALTIME,
```

Comment:
This was previously 30 mins, but API response indicates that it's only available for realtime.